src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadataBuilder.cs

```diff
@@ -101,8 +101,13 @@ public IPackageSearchMetadata Build()
                 IsListed = _metadata.IsListed,
                 PrefixReserved = _metadata.PrefixReserved,
                 LicenseMetadata = _metadata.LicenseMetadata,
-                LazyDeprecationFactory = _lazyDeprecationFactory ?? AsyncLazy.New(_metadata.GetDeprecationMetadataAsync)
-            };
+                LazyDeprecationFactory = _lazyDeprecationFactory ?? AsyncLazy.New(_metadata.GetDeprecationMetadataAsync),
+                LocalPackageInfo = _metadata is LocalPackageSearchMetadata
+                        ? (_metadata as LocalPackageSearchMetadata).LocalPackageInfo
+                        : _metadata is ClonedPackageSearchMetadata
+                        ? (_metadata as ClonedPackageSearchMetadata).LocalPackageInfo
+                        : null
```

Comment:
I don't know if the compiler is smart enough to avoid doing typecasting twice, but there's a chance is will be done once for the "is" check and once again for the "as" conversion.

Also, a repeat from a previous comment, don't use `as` unless you expect times when the type does not match. Given it's immediately after an `is` check, just use normal typecasting.

You could possibly use `= (_metadata as LocalPackageSearchMetadata)?.LocalPackageInfo ?? (_metadata as ClonedPackageSearchMetadata)?.LocalPackageInfo`. However, at this stage we should consider if there's a leaky abstraction or another design issue. `ClonedPackageSearchMetadata` is used for HTTP sources, right? and `LocalPackageSearchMetadata` is for local sources? Well, there are no other types of sources, so you may as well add the property to the base class, so you don't need any type checks or typecasting here.

Reply:
> I don't know if the compiler is smart enough to avoid doing typecasting twice, but there's a chance is will be done once for the "is" check and once again for the "as" conversion.

Me neither. I will assume there's no optimization.

> Also, a repeat from a previous comment, don't use as unless you expect times when the type does not match. Given it's immediately after an is check, just use normal typecasting.

Ok, will do that next time 

> You could possibly use = (_metadata as LocalPackageSearchMetadata)?.LocalPackageInfo ?? (_metadata as ClonedPackageSearchMetadata)?.LocalPackageInfo. However, at this stage we should consider if there's a leaky abstraction or another design issue. ClonedPackageSearchMetadata is used for HTTP sources, right? and LocalPackageSearchMetadata is for local sources? Well, there are no other types of sources, so you may as well add the property to the base class, so you don't need any type checks or typecasting here.

The possible design issue is that  when `IPackageSearchMetatdata`-based object are cloned in `PackageSearchMetadataBuilder`, those objects are transformed
internally to `ClonedPackageSearchMetadata`. That clone only contains the essential info needed in in the `IPackageSearchMetatdata` interface. Specialized
information for each specific info in each of the derived objects are lost. That's why I added a new field in `ClonedPackageSearchMetadata`.